pkg/controller/operators/catalog/step.go

```diff
@@ -84,6 +88,10 @@ func (b *builder) create(step v1alpha1.Step) (Stepper, error) {
 		case crdlib.V1Version:
 			return b.NewCRDV1Step(b.opclient.ApiextensionsInterface().ApiextensionsV1(), &step, manifest), nil
 		case crdlib.V1Beta1Version:
+			// Emit error about deprecated API
+			// This codepath will be removed entirely in 1.22
+			metrics.EmitInstallPlanWarning(v1beta1DeprecationWarning, step.Resource.Name, "",
```

Comment:
Why is this special case necessary?

Reply:
We install CRDs with OLM's client and the scoped client RoundTripper would not apply